src/terraingenerator/main.cpp

```diff
@@ -73,28 +89,132 @@ static void showVersion()
                << qPrintable(QCoreApplication::applicationVersion());
 }
 
-static void parseCommandLineArguments(CommandLineOptions &options)
+static bool parseCommandLineArguments(CommandLineOptions &options)
 {
     const QStringList arguments = QCoreApplication::arguments();
+    bool inPriority = false;
+    bool inSource = false;
+    bool inCombine = false;
+    int argCount = 0;
+    QStringList combine;
+
 
     for (int i = 1; i < arguments.size(); ++i) {
         const QString &arg = arguments.at(i);
+
+        // Process source option.
+        if (inSource) {
+            if (arg.at(0) == QLatin1Char('-')) {
+                // Some other option is starting here.
+                if (argCount == 0) {
+                    qWarning() << "Missing arguments for --input option.";
+                }
+                inSource = false;
+            } else {
+                // Append the source file.
+                argCount++;
+                options.sources.append(arg);
+                //qDebug() << "adding source" << arg;
+            }
+
+            // If we should still process input list, skip the rest of the loop.
+            if (inSource) continue;
```

Comment:
Rather than doing a condition here, just put the `continue` right after `inSource = false` (same for `inPriority` and `inCombine` below).